### PR TITLE
final changes

### PR DESCRIPTION
**Pull request by Mukumbuta**
# About this PR:

### Home page
- When the page loads, the web app shows the list of animals that can be filtered out by any character contained in the names of the animals
- Along with the category name, some numeric values that come from the API are displayed per category
- When the user clicks (or taps) on a category item, the application navigates to the details page

### Details page
- In the details page, the web app retrieves data from the API to show detailed data in the category.
- When the user clicks on the "Back" button (<), the user navigates to the home page

### Testing requirements
- Created unit tests for pure functions in the Redux code
- Created integration tests for the application using the React Testing Library
- Mocked the access to the API, so that the tests don't send actual requests

### Technical requirements
- The project is a single-page application (SPA) built with React and Redux 👍🏾
- The data retrieved from the API should be stored in the Redux store 👍🏾
- You should filter the data that you retrieve from the API using a Filter stateless component 👍🏾
- Every page (the main page and the pages for each item) should have a unique route within the SPA 👍🏾
- The project should be deployed and accessible online (add a link to the online version of your app to your README file) 👍🏾

### Project documentation
- Record a video presenting the features of the project you built 👍🏾

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,105 @@
...
+>Feel free to check the [issues page](../../issues/).
+
+## Show your support
+
+>Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- This project built in Microverse's  Module 3 Capstone Project week.
+- Nelson Sakwa on whose original design this project is based.
```
Comment (OLIPLICHE) [inline README.md]: - You did a good Job, You mentioned Nelson Sakwa. Now you need to Link it to the main website design. 
- I Provide the Link in acknowledging Nelson Sakwa. here is the link:  👉 [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).

File: README.md [hunk 2]
```diff
@@ -1 +1,105 @@
...
+```
+npm install
+```
+
+- Run the tests: 
+```
+npm test
+```
+- or
+```
```
Comment (OLIPLICHE) [inline README.md]: - **Optional**: Kindly remove this dot of lines to make the sections below look better and more visible. 
- **Optional**: Also the images you provided in Home section page and Details page section are broken, Kindly fix that

File: package.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    "react-dom": "^17.0.2",
+    "react-helmet": "^6.1.0",
+    "react-redux": "^7.2.6",
+    "react-router-dom": "^6.2.1",
+    "react-scripts": "^2.1.3",
+    "react-test-renderer": "^17.0.2",
+    "redux": "^4.1.2",
+    "redux-logger": "^3.0.6",
+    "redux-thunk": "^2.4.1"
+  },
```
Comment (OLIPLICHE) [inline package.json:24]: - Please can't run the App. I think you have an issue with your package.json. Kindly check your setup
- I cross-checked at my end. It did not work

File: src/_test_/Animal.test.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import { Provider } from 'react-redux';
+import { BrowserRouter as Router } from 'react-router-dom';
+import store from '../redux/store';
+import Animal from '../component/Animal';
+
+describe('Should render Animal component', () => {
+  test('Should render HomePage', () => {
+    const tree = renderer
+      .create(<Router><Provider store={store}><Animal /></Provider></Router>)
+      .toJSON();
```
Comment (OLIPLICHE) [inline src/_test_/Animal.test.js:12]: - Test 
- 
Capture1

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @Mukumbuta, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit have a green check mark (✅)) 👏
- Commit messages are meaningful 👏 👏
- Gitflow is correctly used (Pull Request is made from the feature branch into the dev branch) 👏

## Required Changes ♻️
- Please make sure the video contains a highlight about a **piece of the code** that you find interesting or are proud of making.
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 5]
```diff
@@ -1 +1,193 @@
...
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+This project is based on the work of [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) on whose original design this project is based.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE) licensed.
```
Comment (FrederickMih) [inline README.md:191]: - [Optional] The License link is not working. Let's kindly fix the link :+1:

File: README.md [hunk 6]
```diff
@@ -1 +1,193 @@
...
+- [👥 Authors](#authors)
+- [🔭 Future Features](#future-features)
+- [🤝 Contributing](#contributing)
+- [⭐️ Show your support](#support)
+- [🙏 Acknowledgements](#acknowledgements)
+- [❓ FAQ](#faq)
+- [📝 License](#license)
+
+
+# 📖 [CMS_Bookstore] <a name="about-project"></a>
```
Comment (FrederickMih) [inline README.md:36]: - [Opitonal] Please correct the title of your project.

File: src/API/api.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
+/* eslint-disable camelcase */
+import { v4 as uuidv4 } from 'uuid';
+
+const ANIMALS_FETCHED = 'ANIMALS_FETCHED';
```
Comment (FrederickMih) [inline src/API/api.js:4]: - [Optional] To easily scale your app in the future, it's advisable to scope your action types using Redux [duck pattern](https://medium.com/@matthew.holman/what-is-redux-ducks-46bcb1ad04b7)

File: src/component/Animal.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+      <div id="home_content_cont">
+        {animalList.length <= 1 ? <p>Just a moment please...</p> : (
+          // eslint-disable-next-line
+          animalList.filter((searchItem) => {
+          // eslint-disable-next-line
+          if (AnimalDetails == '') {
+              return searchItem;
+            } if (searchItem.name.toLowerCase().includes(AnimalDetails.toLowerCase())) {
+              return searchItem;
+            }
```
Comment (FrederickMih) [inline src/component/Animal.js:38]: - [Optional] Filter works perfectly :+1:  but based on the requirement of this project, this was supposed to be achieved on a component that holds **no state**( **Filter** stateless component).

File: src/component/Animal.js [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+        <input
+          id="search-input"
+          type="text"
+          placeholder="Search"
+          onChange={(e) => setAnimalDetails(e.target.value)}
+        />
+      </div>
+      <div id="home_content_cont">
+        {animalList.length <= 1 ? <p>Just a moment please...</p> : (
+          // eslint-disable-next-line
```
Comment (FrederickMih) [inline src/component/Animal.js:31]: - [Optional] Please let's fix all the Linter issues across codebase. Linters flag bugs that are required to fix in order to keep your codebase at standard.

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand:  @Mukumbuta, 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights :green_circle: 

- Great app design :+1: 
- Descriptive README.md file :+1: 
- PR is professional :+1: 
- All Linter checks passed :heavy_check_mark: 

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**